diesel/src/type_impls/integers.rs

```diff
@@ -88,3 +88,85 @@ impl<DB: Backend> ToSql<sql_types::BigInt, DB> for i64 {
             .map_err(|e| Box::new(e) as Box<Error + Send + Sync>)
     }
 }
+
+impl<DB: Backend<RawValue = [u8]>> FromSql<sql_types::SmallInt, DB> for u16 {
```

Comment:
This should only be implemented for `Unsigned<SmallInt>`.

Reply:
Also all of these implementations are backend specific, they should not be generic like this.